modules/apps/object/object-service/src/main/java/com/liferay/object/internal/workflow/ObjectEntryWorkflowHandler.java

```diff
@@ -67,6 +73,35 @@ public String getType(Locale locale) {
 		return _objectDefinition.getLabel(locale);
 	}
 
+	@Override
+	public WorkflowDefinitionLink getWorkflowDefinitionLink(
+			long companyId, long groupId, long classPK)
+		throws PortalException {
+
+		ObjectEntry objectEntry = _objectEntryLocalService.getObjectEntry(
+			classPK);
+
+		WorkflowDefinitionLink workflowDefinitionLink =
+			_workflowDefinitionLinkLocalService.fetchWorkflowDefinitionLink(
+				companyId, groupId, ObjectEntryFolder.class.getName(),
+				objectEntry.getObjectEntryFolderId(),
+				objectEntry.getObjectDefinitionId(), true);
+
+		if (workflowDefinitionLink == null) {
+			workflowDefinitionLink =
+				_workflowDefinitionLinkLocalService.fetchWorkflowDefinitionLink(
+					companyId, groupId, ObjectEntryFolder.class.getName(),
+					objectEntry.getObjectEntryFolderId(),
+					ObjectDefinitionConstants.OBJECT_DEFINITION_ID_ALL, true);
+		}
+
+		if (workflowDefinitionLink != null) {
+			return workflowDefinitionLink;
+		}
+
+		return super.getWorkflowDefinitionLink(companyId, groupId, classPK);
```

Comment:
Can you reverse the condition here? That way, all checks will be the the same (foo != null) and the code it will be easier to scan.
```
WorkflowDefinitionLink workflowDefinitionLink = ...;

if (workflowDefinitionLink != null) {
    return ...;
}

workflowDefinitionLink = ...;

if (workflowDefinitionLink != null) {
   return ...;
}

return super.getWorkflowDefinitionLink(...);
```

Reply:
thanks for pointing at it

done and rebased